service/src/main/java/greencity/service/NotificationServiceImpl.java

```diff
@@ -0,0 +1,146 @@
+package greencity.service;
+
+import greencity.dto.notification.*;
+import greencity.entity.Notification;
+import greencity.entity.User;
+import greencity.enums.NotificationType;
+import greencity.repository.NotificationRepo;
+import greencity.repository.UserRepo;
+import jakarta.persistence.EntityNotFoundException;
+import lombok.RequiredArgsConstructor;
+import org.modelmapper.ModelMapper;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+/**
+ * Service implementation for managing user notifications.
+ */
+@Service
+@RequiredArgsConstructor
+public class NotificationServiceImpl implements NotificationService {
+    private final NotificationRepo notificationRepo;
+    private final UserRepo userRepository;
+    private final ModelMapper modelMapper;
+
+    /**
+     * Creates a notification when a new comment is added.
+     *
+     * @param dto DTO with info about the new comment notification
+     */
+    @Override
+    public void createNewCommentNotification(NewCommentNotificationDto dto) {
+        createNotification(dto.receiver().getId(), NotificationType.NEW_COMMENT, dto.objectId());
+    }
+
+    /**
+     * Creates a notification when a reply is added.
+     *
+     * @param dto DTO with info about the new reply notification
+     */
+    @Override
+    public void createNewReplyNotification(NewReplyNotificationDto dto) {
+        createNotification(dto.receiver().getId(), NotificationType.NEW_REPLY, dto.objectId());
+    }
+
+    /**
+     * Creates a notification when a like is received.
+     *
+     * @param dto DTO with info about the new like notification
+     */
+    @Override
+    public void createNewLikeNotification(NewLikeNotificationDto dto) {
+        createNotification(dto.receiver().getId(), NotificationType.NEW_LIKE, dto.objectId());
+    }
+
+    private void createNotification(Long receiverId, NotificationType type, long objectId) {
+        User receiver = findUserById(receiverId);
+
+        Notification notification = Notification.builder()
+                .type(type)
+                .objectId(objectId)
+                .receiver(receiver)
+                .isRead(false)
+                .build();
+
+        notificationRepo.save(notification);
+    }
```

Comment:
we can use `ModelMapper` for that and have a single method taking any NotificationDto

```suggestion
    public void createNotification(NotificationDto dto) {
        Notification notification = modelMapper.convert(dto, Notification.class);
        notification.setReceiver(findUserById(dto.receiver().getId());
        notificationRepo.save(notification);
    }
```